### PR TITLE
New release cadence documentation

### LINKED ISSUE #977: Document new release cadence
Some of the aspects we should formalise in a document:

- compatibility promisses
  - general aspects of compatibility and semver
  - for the config files it should be based on what Kubernetes dictates 
- feature deprecation policy (probably based on Kubernetes, but could be more relaxed pre-1.0)
- change in default Kubernetes version we ship should trigger minor version change
- how does change in node AMIs affect version number (?)

### PR DESCRIPTION
**Pull request by gemagomez**
Fixes #977

### Description

New release process documentation added to README.md.

### Checklist

- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,17 @@ To learn more abut what `eksctl` can do check [eksctl.io](https://eksctl.io). A
 
 Code contributions are very welcome. If you are interested in helping make `eksctl` great then see our [contributing guide](CONTRIBUTING.md).
 
+## Releases
+
+`eksctl` will be releasing on a 2 week cadence (minor releases) with extra patch releases in between as needed.
```
Comment (errordeveloper) [inline README.md]: I think we can avoid having to change it after 0.2.0 is out... How about something like this:
```
As of eksctl 0.2.0, minor releases should be expected every two weeks and patch release will be made available as soon as bugs get fixed. One or more release candidate(s) will be made available prior to each minor release.
```

Comment (errordeveloper) [reply README.md]: I think one paragraph should be sufficient here.

File: README.md [hunk 2]
```diff
@@ -104,6 +104,17 @@ To learn more abut what `eksctl` can do check [eksctl.io](https://eksctl.io). A
...
+
+This versioning scheme will be effective from version 0.2.0 onwards.
+
+RC releases will be published for testing purposes 2 working days prior to every minor release. At the end of this testing period, and after any regressions have been found and fixed, the release will take place.
+
+It is highly discouraged to use RC releases in any production environment. They are for testing purposes only.
```
Comment (errordeveloper) [inline README.md]: We don't have any other recommendations on what one should or shouldn't use in production, let's remove this statement.

File: README.md [hunk 3]
```diff
@@ -104,6 +104,17 @@ To learn more abut what `eksctl` can do check [eksctl.io](https://eksctl.io). A
...
+
+
```
Comment (errordeveloper) [inline README.md]: Let's keep it single blank line please...

File: README.md (hunk 1 above)
Comment (gemagomez) [reply README.md]: Agreed.

File: README.md (hunk 2 above)
Comment (gemagomez) [reply README.md]: I feel it is important for users to know this is a test build, not a production one. We don't have a disclaimer?

File: README.md (hunk 3 above)
Comment (gemagomez) [reply README.md]: +1

File: README.md (hunk 2 above)
Comment (errordeveloper) [reply README.md]: True. As you suggested on slack, let's just state RCs are only for testing.

**Comment by errordeveloper** [comment]
Will re-open from another branch, turns out that I cannot push to your master branch...

**Pull request closed without merging**